tbx/core/utils/models.py

```diff
@@ -133,3 +133,56 @@ def theme_class(self):
 
     class Meta:
         abstract = True
+
+
+class DivisionMixin(models.Model):
+    """
+    Provides a 'division' field to allow pages to be associated to a Division.
+    """
+
+    division = models.ForeignKey(
+        "divisions.DivisionPage",
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+
+    def get_division(self):
```

Comment:
This will need some good test coverage:
- one with a page directly linked to a division
- one with parent or grandparent linked to a division
- one without link nor any ancestor linked to a division

Reply:
That's a good shout, thank you! Added in [1cd3c0c](https://github.com/torchbox/torchbox.com/pull/335/commits/1cd3c0c09c949b7c04d0a54dd85ef52f3ca16d12)